carma_wm/include/carma_wm/WMTestLibForGuidance.h

```diff
@@ -0,0 +1,458 @@
+  lanelet::routing::RoutingGraphUPtr map_graph = lanelet::routing::RoutingGraph::build(*cmw->getMap(), *traffic_rules);
+
+  // Generate route
+  lanelet::Optional<lanelet::routing::Route> optional_route;
+  std::vector<lanelet::ConstLanelet> middle_lanelets;
+  if (lanelets.size() > 2)
+  {
+    middle_lanelets = std::vector<lanelet::ConstLanelet>(lanelets.begin() + 1, lanelets.end() - 1);
+    optional_route = map_graph->getRouteVia(lanelets[0], middle_lanelets, lanelets[lanelets.size() - 1]);	
+    for (auto llt : middle_lanelets)
+    {
+      std::cerr<< "Id: " << llt.id() << " ";
+    }
+    std::cerr << "=========ended" << std::endl;
+  }
+  else if (lanelets.size() == 2)
+  {
+    optional_route = map_graph->getRoute(lanelets[0], lanelets[1]);
+  }
+  else
+  {
+    std::cerr << "here 1" << std::endl;
+    throw lanelet::InvalidInputError("Fewer than 2 lanelets have been passed to setRouteByLanelets, which cannot be a valid route");
+  }
+  
+  if (!optional_route)
+  {
+    std::cerr << "here 2" << std::endl;
+    
+    throw lanelet::InvalidInputError("There was an error in routing using the given lanelets in setRouteByLanelets");
+  }
+  std::cerr << "heres 1" << std::endl;
+  lanelet::routing::Route route = std::move(*optional_route);
+  std::cerr << "heres 2" << std::endl;
+
+  carma_wm::LaneletRoutePtr route_ptr = std::make_shared<lanelet::routing::Route>(std::move(route));
+  std::cerr << "heres 1" << std::endl;
+  cmw->setRoute(route_ptr);
+
+  std::cout << "New route has been set successfully!" << std::endl;
+}
+
+/**
+ * \brief Sets the route by series of lanelets id. 
+ *
+ * \param lanelets lanelets that you want the route to follow
+ * \param cmw CARMAWorldModel shared_ptr
+ *
+ * \throw std::invalid_argument if fewer than 2 lanelets passed, or the given lanelets are not routable
+ */
+inline void setRouteByIds (std::vector<lanelet::Id> lanelet_ids, std::shared_ptr<carma_wm::CARMAWorldModel> cmw)
+{
+  std::vector<lanelet::Lanelet> lanelets;
+  for (auto id : lanelet_ids)
+  {
+      lanelets.push_back(cmw->getMutableMap()->laneletLayer.get(id));
+  }
+  setRouteByLanelets(lanelets, cmw);
+}
+
+/**
+ * \brief Gets the CARMAWorldModel for the guidance test map
+ *
+ * \param width width of single lanelet, default is 3.7 meters which is US standard
+ * \param length length of a single lanelet, default is 25 meters to accomplish 100 meters of full lane
+ * \param map_options vector of enum map_options. No input is DEFAULT options, but empty vector is NO_OBSTACLE, NO_SPEED_LIMIT options.
+ *                    if mix and match, provide each one explicitly e.g. NO_OBSTACLE, DEFAULT_SPEED_LIMIT
+ * NOTE: Input 1/4th of full lane length you want to accomplish in length. 
+ */
+inline std::shared_ptr<carma_wm::CARMAWorldModel> getGuidanceTestMap(double width = 3.7, double length = 25.0, std::vector<TestMapOption> map_options = {DEFAULT_OBSTACLE, DEFAULT_SPEED_LIMIT})
```

Comment:
Having a vector of enum options here isn't very extensible, since we could only represent the options as flags. I would suggest making a MapOptions class which contains these flags along with the width and length. So this function just takes in a map options object.